hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateTopLevelDagVisitor.java

```diff
@@ -94,13 +95,14 @@
 import static java.util.Collections.singletonList;
 
 public class CreateTopLevelDagVisitor extends CreateDagVisitorBase<Vertex> {
-
     // TODO https://github.com/hazelcast/hazelcast/issues/20383
-    public static final ExpressionEvalContext MOCK_EEC =
-            new ExpressionEvalContext(emptyList(), new DefaultSerializationServiceBuilder().build());
+    // Note: it is used only in CreateTopLevelDagVisitor, and initialized here.
+    // We don't want to serialize it, but we initialize it during the runtime.
+    @SuppressWarnings({"checkstyle:StaticVariableName", "checkstyle:VisibilityModifier"})
+    public static ExpressionEvalContext MOCK_EEC;
```

Comment:
It's initialized in the constructor, it could be overwritten by concurrent queries, and also by multiple HZ instances on the same JVM!

It should be a private non-static field.